From 6e9019984225147ba0e9c2cf018068e635dd4d99 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Sun, 21 Feb 2021 21:00:38 +0530
Subject: [PATCH] libmedia: Add MediaPlayer() function for backward
 compatibility

 * package name argument got added to MediaPlayer in fd90fdfe2a611ae824a32e236da288b1fdfd445d, which
   broke most prebuilt libcameraservice as they were built before
   this commit was merged

 * although a shim for this exists, it doesn't work well and
   crashes cameraservice whenever MediaPlayer() is called, seemingly
   due to a stack corruption

Change-Id: I1f8e112b1ab9cc96cb877dbee14767c28a377ab1
---
 media/libmedia/include/media/mediaplayer.h | 3 ++-
 media/libmedia/mediaplayer.cpp             | 4 ++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/media/libmedia/include/media/mediaplayer.h b/media/libmedia/include/media/mediaplayer.h
index 7c29e50d8c..bb0d8adfce 100644
--- a/media/libmedia/include/media/mediaplayer.h
+++ b/media/libmedia/include/media/mediaplayer.h
@@ -207,7 +207,8 @@ class MediaPlayer : public BnMediaPlayerClient,
                     public virtual IMediaDeathNotifier
 {
 public:
-    MediaPlayer(const std::string opPackageName = "");
+    MediaPlayer();
+    MediaPlayer(const std::string opPackageName);
     ~MediaPlayer();
             void            died();
             void            disconnect();
diff --git a/media/libmedia/mediaplayer.cpp b/media/libmedia/mediaplayer.cpp
index 6079a2dbe1..f1e48ad188 100644
--- a/media/libmedia/mediaplayer.cpp
+++ b/media/libmedia/mediaplayer.cpp
@@ -41,6 +41,10 @@ namespace android {
 
 using media::VolumeShaper;
 
+MediaPlayer::MediaPlayer() : MediaPlayer("" /*opPackageName*/)
+{
+}
+
 MediaPlayer::MediaPlayer(const std::string opPackageName) : mOpPackageName(opPackageName)
 {
     ALOGV("constructor");
-- 
2.25.1

